### ISSUE TITLE
Why does the scan operator only emit the initialValue when receiving the first value from the observable (or on completion)?

### ISSUE THREAD

**Issue by amoerie** [ReactiveX/RxJava#3522]
Hello, I am wondering why the scan() method with an initialValue only emits the initialValue when the first value from the observable arrives. I had expected this to happen instantly upon subscription. 

For now, I worked around this issue by NOT providing an initialValue, and adding ".startWith()" to the observable I am scanning. 

In the `OperatorScan` class:

``` java

        return new Subscriber<T>(child) {
            ...

            @Override
            public void onNext(T currentValue) {
                emitInitialValueIfNeeded(child);
                ...
                child.onNext(this.value);
            }

            @Override
            public void onError(Throwable e) {
                child.onError(e);
            }

            @Override
            public void onCompleted() {
                emitInitialValueIfNeeded(child);
                child.onCompleted();
            }
        }
```
Example scenario: scan an observable that counts from 0 to 4, emitting a number every second.

```java
Observable
    .create(new Observable.OnSubscribe<Integer>() {
        @Override
        public void call(Subscriber<? super Integer> subscriber) {
            for (int i = 0; i < 5; i++) {
                try {
                    Thread.sleep(1000);
                } catch (InterruptedException e) {
                    subscriber.onError(e);
                }
                subscriber.onNext(i);
            }
            subscriber.onCompleted();
        }
    })
    .scan(0, new Func2<Integer, Integer, Integer>() {
        @Override
        public Integer call(Integer sum, Integer nextNumber) {
            return sum + nextNumber;
        }
    })
```

The timeline of this observable would be:

0ms : subscribe()
1000ms: 0 <- the initial value
1001ms: 0 <- the acummulated result of 0 + the initial value
2000ms: 1 <- the accumulated result of 1 + 0
3000ms: 3 <- the accumulated result of 2 + 1
4000ms: 6 <- the accumulated result of 3 + 2
5000ms: 10 <- the accumulated result of 4 + 6

 
My question is: why does this happen lazily. Why not immediately call `child.onNext(initialValue)` when the subscription starts? Why does the given example not produce value 0 at 1 ms? 

Thank you in advance.

**Comment by akarnokd** [comment]
This was a bug with 1.0.14 and has been fixed in 1.0.15. You should, however, switch to 1.0.16 instead as 1.0.15 had a regression bug with null initial value.

**Comment by amoerie** [comment]
Ah ok, I was indeed not running on the latest version, my apologies. That seems to have fixed the problem indeed! Thanks for the quick response.
